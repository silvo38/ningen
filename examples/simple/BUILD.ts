#!/usr/bin/env -S deno run --allow-read --allow-write --unstable
// The shebang above lets you execute this BUILD.ts file directly.

// TODO: Give real deno.land URL.
import { File, init } from "../../mod.ts";

// Initialises Ningen and retrieves the functions for defining rules and build
// targets.
const { build, rule, file, files, generate } = init(import.meta.url);

// Define a Rule for the `append.sh` script.
const appendRule = rule({
  name: "append",
  command: "./append.sh $in $out",
  srcs: files("append.sh"),
});

// Define a helper function that invokes the rule.
function append(src: File) {
  build({
    rule: appendRule,
    inputs: [src],
    outputs: [src.replaceExtension(".out")],
  });
}

// Process a file.
append(file("foo.txt"));

// Write the build.ninja file. Can override the output file if you like.
generate();
