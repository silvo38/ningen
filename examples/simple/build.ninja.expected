rule append
  command = ./append.sh $in $out
  description = Appending text

build foo.out: append foo.txt | append.sh
